pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
         return self._get_statement(code) + '\n'
 
     def _print_PythonPrint(self, expr):
+        end = '\n'
+        sep = ' '
+        code = ''
+        empty_end = FunctionCallArgument(LiteralString(''), 'end')
+        space_end = FunctionCallArgument(LiteralString(' '), 'end')
+        for f in expr.expr:
+            if f.has_keyword:
+                if f.keyword == 'sep':
+                    sep = str(f.value)
+                elif f.keyword == 'end':
+                    end = str(f.value)
+        args_format = []
         args = []
-        n = len(expr.expr)
-        for j, f in enumerate(expr.expr):
+        orig_args = [f for f in expr.expr if not f.has_keyword]
+
+        def formatted_args_to_print(fargs_format, fargs, end):
+
+            def split_long_argument(line):
+                line = line[1 : len(line) - 1]
+                length = len(line)
+                longest_chunk = length
+                cuts = []
+                positions = [idx for idx in range(length) if line[idx] == ' ']
+                if len(positions) > 12:
+                    positions = positions[int(len(positions) / 4) : 12]
```

Comment:
I don't understand what you are doing here. If there are more than 12 spaces in the line then only consider some of the spaces? What happens for a line like:
```
print("a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a")
```
This line contains 52 spaces so after the lines above `positions` is an empty list

Reply:
I don't think it will produce an empty list. I will iterate through all possible subsets of the positions to determine the ideal break. The complexity required will be n * 2 ^ n where n is the size of the positions. So, n cannot be too large. I chose 12 in this case, for a total of about 49000 iterations. this will run in about 4 microseconds even with python. To keep only 12 positions, i thought i would be better off skipping the first quarter of the string since those space characters will be too far off, then i just take the 12 positions from there to be my candidates.

Reply:
Sorry, you are right! i should have `int(len(positions) / 4) + 12` instead of `12`.